### PR TITLE
[small] Fix to check overflow/underflow in V1 aggregators, and throw appropriate exceptions

### PR DESCRIPTION
**Pull request by igor-aptos**
in addition to fixing code, tests had [should_panic] which is incorrect.
I will recheck - but I think these fail on main too, as I think [should_panic] was mistakenly added back in https://github.com/aptos-labs/aptos-core/commit/3e29accd9d9aa2ca4802bcfa35847d13c3af7ccd

### CODE REVIEW

File: aptos-move/e2e-move-tests/src/harness.rs [hunk 1]
```diff
@@ -797,6 +797,51 @@ macro_rules! assert_abort {
...
 
+/// Helper to assert transaction aborts.
+#[macro_export]
+macro_rules! assert_abort_ref {
+    // identity needs to be before pattern (both with and without message),
+    // as if we pass variable - it matches the pattern arm, but value is not used, but overriden.
+    // Opposite order and test_asserts_variable_used / test_asserts_variable_used_with_message tests
+    // would fail
+    ($s:expr, $c:ident $(,)?) => {{
+        claims::assert_matches!(
```
Comment (igor-aptos) [inline aptos-move/e2e-move-tests/src/harness.rs:808]: without using assert_matches (but using assert!(matches!())) we don't get nice error messages.

but assert_matches takes ownership of the object, so we need it to be on references.
will see to dedup on main

Comment (georgemitenkov) [reply aptos-move/e2e-move-tests/src/harness.rs:808]: Should assert_abort always match the references by default?

**Review by georgemitenkov — APPROVED** [review]

File: aptos-move/e2e-move-tests/src/harness.rs [hunk 2]
```diff
@@ -797,6 +797,51 @@ macro_rules! assert_abort {
...
+    }};
+    ($s:expr, $c:pat $(,)?) => {{
+        claims::assert_matches!(
+            $s,
+            &aptos_types::transaction::TransactionStatus::Keep(
+                aptos_types::transaction::ExecutionStatus::MoveAbort { code: $c, .. }
+            )
+        );
+    }};
+    ($s:expr, $c:ident, $($arg:tt)+) => {{
```
Comment (vusirikala) [inline aptos-move/e2e-move-tests/src/harness.rs:824]: The `assert_abort_ref` macro has 4 expressions of type `($s: expr, ...) => {..}`. What does that mean?

File: aptos-move/e2e-move-tests/src/harness.rs (hunk 1 above)
Comment (igor-aptos) [reply aptos-move/e2e-move-tests/src/harness.rs:808]: yup, but callers need to be changed, we have a lot of them.

File: aptos-move/e2e-move-tests/src/harness.rs (hunk 2 above)
Comment (igor-aptos) [reply aptos-move/e2e-move-tests/src/harness.rs:824]: it means, based on what is passed, different variant is instantiated. each of the four branches matches a different thing. you can check unit tests for variety of ways to call it (i.e. with message text and without, etc)

**Review by vusirikala — APPROVED** [review]

File: aptos-move/e2e-move-tests/src/harness.rs (hunk 2 above)
Comment (gelash) [reply aptos-move/e2e-move-tests/src/harness.rs:824]: huh I didn't know, macro power user level ++

**Review by gelash — APPROVED** [review]

**Pull request merged**